tasks/scheduler.go

```diff
@@ -193,15 +206,25 @@ func (s *scheduler) tryInitMultiVersionStore(ctx sdk.Context) {
 	s.multiVersionStores = mvs
 }
 
-func indexesValidated(tasks []*deliverTxTask, idx []int) bool {
-	for _, i := range idx {
+func dependenciesValidated(tasks []*deliverTxTask, deps map[int]struct{}) bool {
+	for i := range deps {
```

Comment:
does the order of this validation matter? map iteration isn't guaranteed in order so if it does we should make a slice + sort first

Reply:
it does not matter because all must be validated to move forward